### ISSUE TITLE
How do you run Electrum mobile on testnet?

### ISSUE THREAD

**Issue by fresheneesz** [spesmilo/electrum#6480]
I was hoping there would be a settings switch to testnet, or at least that there'd be a testnet version of the app I could install. But I don't see either of those things. How do I run testnet on mobile?

**Comment by ecdsa** [comment]
the published APK is for mainnet only.
to use on testnet, you would need to compile your own APK

**Comment by fresheneesz** [comment]
Would you accept a pull request that either added a settings switch to the app for testnet or added a build path for building a testnet version of the APK that could be published?

**Comment by ecdsa** [comment]
I think a runtime switch for testnet would complicate the architecture unnecessarily. The --testnet option is passed from the command line, and it cannot be changed at runtime. In other words, you would need to compile a separate APK for testnet

I also think that testnet is not very useful for experimenting with lightning, because some assumptions about mining and block interval break when you use testnet (mining is very irregular there). Having to deal with those issues would not be nice for us.

**Comment by fresheneesz** [comment]
> I also think that testnet is not very useful for experimenting with lightning

Really? That's very surprising to hear. How do you test with lightning while you're developing? I would actually assume testnet's less robust/consistent properties actually make it *easier* to find edge-case bugs (or just WIP edge cases). Even for those not trying to bug hunt and just testing for themselves, those kinds of edge cases can be very educational. Alternatively, the testnet version could tweak the parameter and expose them for testing so things will work properly (if potentially more slowly) and at the same time allowing people the opportunity to play around with failure modes. For example, I still have never personally tested forced channel closure and I would like to. But I would only be comfortable doing that on testnet. 

Also, lightning isn't even the primary function of Electrum, so its odd to hear the opinion that we would want to keep it harder for users to use testnet just because testnet might be less useful for one feature of the wallet.

> Having to deal with those issues would not be nice for us.

I heard what you're saying. Users would come and write issues complaining about this or that on testnet. I would imagine that's true, but I can't imagine that would be any kind of deluge. If people have to specifically seek out the testnet version on the app store, those people are likely to already be pretty savvy with this kind of stuff.

**Comment by fresheneesz** [comment]
@ecdsa I tried building an Electrum mobile apk after [adding `'testnet': true` here](https://github.com/spesmilo/electrum/blob/master/run_electrum#L315). But when I try to run Electrum, it loads for a bit then silently exits.

**Comment by SomberNight** [comment]
@fresheneesz that should work
it might be crashing because there's already mainnet files there?
Check logcat: https://github.com/spesmilo/electrum/tree/36178df875eef5186b9c25622b7b5ae345776ecf/contrib/android#how-can-i-see-logs-at-runtime

**Comment by fresheneesz** [comment]
Yeah, I didn't have time to check logcat. It took me like 4 days of my (admittedly limited) free time to attempt to figure out how to do this. I moved recently, so I might start having free time again soon to come back to this. But this difficulty is exactly why I think there needs to be a mobile testnet app, so its not so difficult to test things out. Especially if installing a locally compiled application has interactions with the version installed from google play. Would be nice to just be able to install Electrum Testnet and start playing around within 30 seconds rather than 3-6 hours (or more depending on familiarity with these tools). I had especially large trouble because I wanted to try doing this in the linux subsystem on windows, which I had to do a ton of upgrades to get working, and still had issues, etc etc. But thanks, I'll get back to this at some point.
